File: php/dnaAnalysis.php
<?php

/**
 * Loads DNA from the specified file line by line, stripping all
 * whitespace and returning the DNA string.
 */
function loadDnaFromFile() {

  $fileContents = "";
  $fileHandle = fopen("../data/H1N1nucleotide.txt", "r");

  while(!feof($fileHandle)) {
    //get the next line
    $line = fgets($fileHandle);
    //concatenate
    $fileContents .= $line;
  }
  fclose($fileHandle);

  //strip all whitespace
  $dna = preg_replace("/\s*/", "", $fileContents); 
  return $dna; 
}

/**
  * Counts the number of times $subSequence appears in the
  * given $dna string.
  */
function countSubsequences($dna, $subSequence) {
  //TODO: implement
}

$subSequence = NULL;
$dna = loadDnaFromFile();
$count = countSubsequences($dna,$subSequence);

printf("%s appears %d times\n", $subSequence, $count);

?>

File: python/dna_analysis.py
import sys
import re

def load_dna_from_file(fileName):
    f = open(fileName, "r")
    # read all lines into a list
    fileLines = f.readlines()
    f.close()
    # list comprehension to strip all whitespace
    strippedLines = [re.sub(r'\s*', '', line) for line in fileLines]
    # join all strings together
    dna = "".join(strippedLines)
    return dna

def count_subsequences(dna, subsequence):
    # TODO: implement

def main():
    # TODO: check the correct number of CLAs
    dna = load_dna_from_file("../data/H1N1nucleotide.txt")

    # TODO: change this to the "first" CLA
    subsequence = None
    count = count_subsequences(dna, subsequence)
    print(f"{subsequence} appears {count} times");

if __name__ == "__main__":
    main()
